WebSocket/src/taosResult.ts

```diff
@@ -0,0 +1,265 @@
+import { AppendRun } from "./ut8Helper"
+
+export class TaosResult {
+    meta: Array<ResponseMeta> | null;
+    data: Array<Array<any>> | null;
+    precision: number = 0;
+    affectRows: number = 0;
+    timing: bigint;
+    constructor(queryResponse: WSQueryResponse) {
+        if (queryResponse.is_update == true) {
+            this.meta = null
+            this.data = null
+        } else {
+            if (queryResponse.fields_count && queryResponse.fields_names && queryResponse.fields_types && queryResponse.fields_lengths) {
+                let _meta = [];
+                for (let i = 0; i < queryResponse.fields_count; i++) {
+                    _meta.push({
+                        name: queryResponse.fields_names[i],
+                        type: queryResponse.fields_types[i],
+                        length: queryResponse.fields_lengths[i]
+                    })
+                }
+                this.meta = _meta;
+            } else {
+                throw new TaosResultError(`fields_count,fields_names,fields_types,fields_lengths of the update query response should be null`)
+            }
+            this.data = [];
+        }
+
+        this.affectRows = queryResponse.affected_rows
+        this.timing = queryResponse.timing
+        this.precision = queryResponse.precision
+    }
+
+    setRows(fetchResponse: WSFetchResponse) {
+        this.affectRows += fetchResponse.rows;
+        this.timing = this.timing + fetchResponse.timing
+    }
+
+    setData(fetchBlockResponse: WSFetchBlockResponse) {
+        if (this.data) {
+            this.data.push([])
+        } else {
+            throw new TaosResultError(`update query response cannot set data`)
+        }
+
+    }
+    /**
+     * Mapping the WebSocket response type code to TDengine's type name. 
+     */
+    getTDengineMeta(): Array<TDengineMeta> | null {
+        if (this.meta) {
+            let _ = new Array<TDengineMeta>()
+            this.meta.forEach(m => {
+                _.push({
+                    name: m.name,
+                    type: TDengineTypeName[m.type],
+                    length: m.length
+                })
+            })
+            return _;
+        } else {
+            return null;
+        }
+    }
+
+}
+
+interface TDengineMeta {
+    name: string,
+    type: string,
+    length: number,
+}
+interface ResponseMeta {
+    name: string,
+    type: number,
+    length: number,
+}
+
+export function parseBlock(fetchResponse: WSFetchResponse, blocks: WSFetchBlockResponse, taosResult: TaosResult): TaosResult {
+
+    if (taosResult.meta && taosResult.data) {
+        let metaList = taosResult.meta;
+
+        // console.log(typeof taosResult.timing)
+        // console.log(typeof blocks.timing)
+        // console.log(blocks.id)
+
+        taosResult.timing = BigInt(taosResult.timing).valueOf() + blocks.timing;
+
+        const INT_32_SIZE = 4;
+
+        let bufferOffset = (4 * 5) + 8 + (4 + 1) * metaList.length;
```

Comment:
could you please add a few lines comments to describe what means about these numbers?